src/universal_transfer_operator/data_providers/base.py

```diff
@@ -37,33 +34,8 @@ class DataProviders(ABC, Generic[DatasetType]):
     def __init__(
         self,
         dataset: DatasetType,
-        transfer_mode,
-        transfer_params: TransferIntegrationOptions = TransferIntegrationOptions(),
     ):
         self.dataset: DatasetType = dataset
-        self.transfer_params = transfer_params
-        self.transfer_mode = transfer_mode
-        self.transfer_mapping: set[Location] = set()
-        self.LOAD_DATA_NATIVELY_FROM_SOURCE: dict = {}
-
-    def __repr__(self):
-        return f'{self.__class__.__name__}(conn_id="{self.dataset.conn_id})'
-
-    @property
-    def hook(self) -> BaseHook:
```

Comment:
Why is this removed from the base dataprovider?

Reply:
Since we are adding dataframe dataset, which will only be in memory construct, now that we are catering to in-memory objects only, having hook, openlineage_dataset_namespace, openlineage_dataset_name, and openlineage_dataset_uri don't make sense to me. What I think should be done is create a base class free of all the destination-specific and have a structure-like. WDYT?


`Dataset`
|__> `volatileDatasets`
          |__> `Dataframe`
|__>  `NonVolatileDatasets 
          |__> `TableDataset`
          |__> `FileDataset`